sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -719,8 +719,7 @@ case class GpuLike(left: Expression, right: Expression, escapeChar: Char)
     } else {
       null
     }
-    val regexStr = escapeLikeRegex(likeStr, escapeChar)
-    lhs.getBase.matchesRe(regexStr)
+    lhs.getBase.like(Scalar.fromString(likeStr), Scalar.fromString(Character.toString(escapeChar)))
```

Comment:
The scalars are leaked here. We need to be sure that we can close them, also GpuScalar already holds a Scalar so we can use it directly.

```
override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
  withResource(Scalar.fromString(Character.toString(escapeChar))) { escapeScalar =>
    lhs.getBase.like(rhs.getBase, escapeScalar)
  }
}
```

Reply:
Updated. Thank you very much!